### PR TITLE
Enable optimizations with JProfiling

### PR DESCRIPTION
**Pull request by ncough**
This change enables various optimizations
that were disabled under JitProfiling due
to issues with the duplication it performed,
such as the increased node count or its
limitations on asyncchecks.

### CODE REVIEW

**Comment by andrewcraik** [comment]
LGTM from me - over to @vijaysun-omr for a review since this is probably in his area of expertise

File: compiler/optimizer/CompactLocals.cpp [hunk 1]
```diff
@@ -90,6 +90,9 @@ int32_t TR_CompactLocals::perform()
       return 0;
       }
 
+   if (comp()->getProfilingMode() == JitProfiling)
```
Comment (vijaysun-omr) [inline compiler/optimizer/CompactLocals.cpp]: Unlike the other changes in this commit this seems to be a new optimization that has been disabled under JitProfiling (?)

**Comment by vijaysun-omr** [comment]
genie-omr build all

File: compiler/optimizer/CompactLocals.cpp (hunk 1 above)
Comment (ncough) [reply compiler/optimizer/CompactLocals.cpp]: This optimization was disabled using strategy flags in the past. The initial version of this commit shifted these checks to the optimizations themselves, as they weren't part of the strategy, rather they were a restriction due to the profiling implementation.

The change is being reverted to use strategy flags, as this makes the disable clearer.

**Comment by ncough** [comment]
This change now enables optimizations under JProfiling through the use of optimization flags, in particular `IfNotJitProfiling` and `IfEnabledAndNotJitProfiling`.

**Comment by andrewcraik** [comment]
The travis failure seems odd - 10 mins inactivity?

I prefer this approach and think this looks good.

**Comment by andrewcraik** [comment]
@vijaysun-omr over to you

**Comment by vijaysun-omr** [comment]
This commit does look much more straightforward. However since there are significant changes, I will do another genie build.

**Comment by vijaysun-omr** [comment]
genie-omr build all

**Comment by vijaysun-omr** [comment]
Merging since the CI issue not a real one.

**Pull request merged**